File: packages/insomnia-httpsnippet/test/fixtures/output/javascript/xhr/text-plain.js
var data = "Hello World";

var xhr = new XMLHttpRequest();
xhr.withCredentials = true;

xhr.addEventListener("readystatechange", function () {
  if (this.readyState === this.DONE) {
    console.log(this.responseText);
  }
});

xhr.open("POST", "http://mockbin.com/har");
xhr.setRequestHeader("content-type", "text/plain");

xhr.send(data);

File: packages/insomnia-httpsnippet/test/fixtures/output/ocaml/cohttp/text-plain.ml
open Cohttp_lwt_unix
open Cohttp
open Lwt

let uri = Uri.of_string "http://mockbin.com/har" in
let headers = Header.add (Header.init ()) "content-type" "text/plain" in
let body = Cohttp_lwt_body.of_string "Hello World" in

Client.call ~headers ~body `POST uri
>>= fun (res, body_stream) ->
  (* Do stuff with the result *)
